src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+  @Override
+  public abstract String comment();
+
+  public static Builder builder() {
+    return new AutoValue_JavaDocComment.Builder();
+  }
+
+  public void accept(AstNodeVisitor visitor) {
+    visitor.visit(this);
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    String throwsType = null;
+    String throwsDescription = null;
+    String deprecated = null;
+    List<String> paramsList = new ArrayList<>();
+    List<String> componentsList = new ArrayList<>();
+    // Private accessor, set complete and consolidated comment.
+    abstract Builder setComment(String comment);
+
```

Comment:
Use local variables here for all components, since we will not call them outside of the class. `Builder.build()` formats and groups the components and sets them as a consolidated comment in `setComment()`. Just like `LineComment` and `BlockComment` classes, we can directly call `comment()` to get the complete comment for `JavaDocComment`.

Reply:
Structure LGTM. Can we please add a comment to these local vars, that the lack of a getter in the external class is WAI (working as intended)?